### PR TITLE
D205 Support - Provider: Google

### PR DESCRIPTION
**Pull request by ferruzzi**
Part of https://github.com/apache/airflow/issues/10742

D205 asserts that all docstrings must have a one-line summary ending in a period.  If there is more than one sentence then there must be a blank line before the rest of the docstring.  Meeting these requirements could be as simple as adding a newline, or might require some rephrasing.

There are almost a thousand violations in the repo so we're going to have to take this in bites.

### PLEASE NOTE

There should be zero logic changes in this PR, only changes to docstrings and whitespace.  If you see otherwise, please call it out.

### Included in this chunk

All files contained in the Google provider package

### To test

If you comment out [this line](https://github.com/apache/airflow/blob/main/pyproject.toml#L68) and run pre-commit in main you will get around 642 errors.  After these changes, "only" 359 remain and no files in any of the above provider packages should be on the list.  After uncommenting that line and rerunning pre-commits, there should be zero regressions. 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Some build docs failure.

**Comment by ferruzzi** [comment]
> Some build docs failure.

Yeah, I saw those locally but I'm not sure what they mean yet.  I'll need to poke at them tonight I guess unless it's obvious to someone else.

**Comment by potiuk** [comment]
Obviously those are blank indentation issues in docs generated via autoapi that are not-too-obvious where they are coming from as the autoapi+sphinx are known to speak in riddles. 

You might have to take a look at generated .rst files in `_api` folder and hope that you will find out quickly. Other than that, I sometimes had to revert to bisecting to narrow down similar problems and remove parts of my modifications to be able to pin-point the places they were coming from,

Usually it SOME problem with identation that docstrings processesed by Autoapi make Sphinx very unhappy about.

**Comment by potiuk** [comment]
(and sorry I can't be more helpful :).

**Comment by ferruzzi** [comment]
Thanks for the tip.   Sometimes it's one of those things where someone else will look and say "on, there's an extra space there at the beginning of line 20" or something.  I was hoping it was one of those days :P     I'll start poking at it this evening and see what I cans see.

File: airflow/providers/google/common/hooks/base_google.py [hunk 1]
```diff
@@ -497,8 +488,11 @@ def wrapper(self: GoogleBaseHook, *args, **kwargs):
     @contextmanager
     def provide_gcp_credential_file_as_context(self) -> Generator[str | None, None, None]:
         """
-        Context manager that provides a Google Cloud credentials for application supporting `Application
-        Default Credentials (ADC) strategy <https://cloud.google.com/docs/authentication/production>`__.
+        Provides a Google Cloud credentials for Application Default Credentials (ADC) strategy support.
+
+        See:
+            `Application Default Credentials (ADC)
+            strategy <https://cloud.google.com/docs/authentication/production>`__.
```
Comment (uranusjr) [inline airflow/providers/google/common/hooks/base_google.py:495]: Use `.. seealso::` for this

**Comment by uranusjr** [comment]
Docstrings in the Google provider is a tower of cards that collapse if you touch it. I don’t have anything good to say about it except good luck… splitting the PR into 10 could be a good idea to isolate the issue (although it may not help you understand _why_ there’s an issue).

**Comment by ferruzzi** [comment]
> splitting the PR into 10

I may do just that.  I was trying to keep it in reasonable chunks rather than flooding a hundred PRs but I think you are right on this one.  I'll try to get to it tonight, I didn't have time last night to look into it.

**Comment by ferruzzi** [comment]
Alright, everything is passing locally now, let's see if I did the right magic incantation this time.

File: airflow/providers/google/cloud/operators/dataflow.py [hunk 2]
```diff
@@ -1018,12 +1018,12 @@ class DataflowCreatePythonJobOperator(GoogleCloudBaseOperator):
     """
     Launching Cloud Dataflow jobs written in python.
 
-     Note that both dataflow_default_options and options will be merged to specify pipeline
```
Comment (ferruzzi) [inline airflow/providers/google/cloud/operators/dataflow.py]: There was a leading space in that line which was causing the indent error. 🤦    That was like playing "find the misplaced semicolon" LOL

Comment (potiuk) [reply airflow/providers/google/cloud/operators/dataflow.py]: BEEN THERE DONE THAT.

**Pull request merged**